configurationproperties-annotationprocessor/src/main/java/com/vaadin/flow/spring/ConfigurationPropertiesAnnotationProcessor.java

```diff
@@ -300,38 +283,62 @@ private void writePostConstruct(PrintWriter out,
                                     List<FieldDescriptor> fields) {
         out.println(" @PostConstruct");
         out.println(" public void transferValuesToSystem() {");
-        fields.forEach(f -> {
-            out.println("  if (" + f.fieldName + " != null) " +
-                    "System.setProperty(\"vaadin." + f.formerPropertyName
-                    + "\", " + f.fieldName + ".toString());");
+        fields.forEach(field -> {
+            String code = "  if (" + field.fieldName + " != null) " +
+                    "System.setProperty(\"vaadin." + field.formerPropertyName
+                    + "\", " + field.fieldName + ".toString());";
+            if (isPrimitive(field)) {
+                code = "  System.setProperty(\"vaadin." + field.formerPropertyName
+                        + "\", \"\" + " + field.fieldName + ");";
+            }
+            out.println(code);
         });
         out.println(" }");
     }
 
+    private boolean isPrimitive(FieldDescriptor f) {
+        boolean primitive = boolean.class.toString().equals(f.type)
```

Comment:
There are more primitive types.
`Class.isPrimitive` does what you need.

Reply:
ok. I didn't remember that method